File: f072(garden)/C++/garden.cpp
#include <iostream>
#include <vector>
int main(int argc, char const *argv[]) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);

  int num_of_fields;
  while (std::cin >> num_of_fields) {
    int fields[num_of_fields] = {0};
    for (int &f: fields) std::cin >> f;
    int start = 0, end = 0, flowers = 0;
    std::vector<int> fence_position;
    int num_of_fences = 0;
    for (int i = 0; i < num_of_fields; i++) {
      if (fields[i] == 1) {
        fence_position.push_back(i);
        num_of_fences++;
      }
    }

    int i = 0;
    while (i < num_of_fences - 1) {
      start = fence_position.at(i);
      end = fence_position.at(i + 1);
      int j = start + 1;
      while (j < end) {
        if (fields[j] == 0 && fields[j + 1] != 9 && fields[j - 1] != 9)
          flowers++;
        j++;
      }
      i++;
    }
    std::cout << flowers << '\n';
  }
  return 0;
}

File: f072(garden)/Python/garden.py

while True:
    try:
        num_of_fields = int(input())
        fields = list(map(int, input().split()))
        flowers = 0
        start, end = 0, 0
        fence_position = [i for i in range(num_of_fields) if fields[i] == 1]
        i = 0
        while i < len(fence_position) - 1:
            start, end = fence_position[i],  fence_position[i + 1]
            j = start + 1
            while j < end:
                if fields[j] == 0 and fields[j + 1] != 9 and fields[j - 1] != 9:
                    flowers += 1
                j += 1
            i += 1
        print(flowers)
    except:
        break
